net-snmp: fix mib representation of timeout values

The patch comes from the follow three commits on upstream:
78dac6e37943d1ad99841898806ea60b0eede636
390303059fbd98b1ee7621ddd4ad4c11d100fff9
96302af7fc3108c208227432f0f0b75f3e7b906d

The first commit:
Fix bug number #a2478: fix mib representation of timeout values.
The second commit:
Reverts 78dac6e37943d1ad99841898806ea60b0eede636 and resolve a2478
in a way that avoids truncating
The third commit:
Corrects conversion factor from microseconds to centiseconds.

Upstream-Status: Backport

Signed-off-by: Per Hallsmark <per.hallsmark@windriver.com>
Signed-off-by: Wenlin Kang <wenlin.kang@windriver.com>

diff -uarN net-snmp-5.7.2-org/agent/mibgroup/notification/snmpNotifyTable.c net-snmp-5.7.2/agent/mibgroup/notification/snmpNotifyTable.c
--- net-snmp-5.7.2-org/agent/mibgroup/notification/snmpNotifyTable.c	2015-04-09 15:24:18.393570318 +0800
+++ net-snmp-5.7.2/agent/mibgroup/notification/snmpNotifyTable.c	2015-04-09 15:25:23.547569858 +0800
@@ -331,7 +331,7 @@
     ptr->tAddressLen = t->remote_length;
     ptr->tAddress = t->remote;
 
-    ptr->timeout = ss->timeout / 1000;
+    ptr->timeout = ss->timeout / 10000;
     ptr->retryCount = ss->retries;
     SNMP_FREE(ptr->tagList);
     ptr->tagList = strdup(ptr->name);
diff -uarN net-snmp-5.7.2-org/agent/mibgroup/target/snmpTargetAddrEntry.h net-snmp-5.7.2/agent/mibgroup/target/snmpTargetAddrEntry.h
--- net-snmp-5.7.2-org/agent/mibgroup/target/snmpTargetAddrEntry.h	2015-04-09 15:24:18.593570085 +0800
+++ net-snmp-5.7.2/agent/mibgroup/target/snmpTargetAddrEntry.h	2015-04-09 15:26:41.250570178 +0800
@@ -51,7 +51,7 @@
          int             tDomainLen;
          unsigned char  *tAddress;
          size_t          tAddressLen;
-         int             timeout;
+         int             timeout;	/* Timeout in centiseconds */
          int             retryCount;
          char           *tagList;
          char           *params;
diff -uarN net-snmp-5.7.2-org/agent/mibgroup/target/target.c net-snmp-5.7.2/agent/mibgroup/target/target.c
--- net-snmp-5.7.2-org/agent/mibgroup/target/target.c	2015-04-09 15:24:18.592569768 +0800
+++ net-snmp-5.7.2/agent/mibgroup/target/target.c	2015-04-09 15:25:52.586569961 +0800
@@ -240,7 +240,7 @@
                             }
 #endif
                             memset(&thissess, 0, sizeof(thissess));
-                            thissess.timeout = (targaddrs->timeout) * 1000;
+                            thissess.timeout = (targaddrs->timeout) * 10000;
                             thissess.retries = targaddrs->retryCount;
                             DEBUGMSGTL(("target_sessions",
                                         "timeout: %d -> %ld\n",
